vocabularies/Hierarchy.json

```diff
@@ -140,6 +155,115 @@
+          { "$Name": "HierarchyNodes", "$Collection": true, "$Type": "Edm.EntityType" },
+          { "$Name": "HierarchyQualifier", "$Type": "Aggregation.HierarchyQualifier" },
+          {
+            "$Name": "Entity",
+            "$Type": "Edm.EntityType",
+            "@Core.Description": "The entity to become a new node",
+            "@Core.LongDescription": "If the entity exists already without a node identifier for the recursive hierarchy,\n            it is upserted with PATCH semantics.\n            To insert an existing entity with key property `ID` as a new node with server-determined node identifier,\n            provide the parameter `\"Entity\": {\"ID\": \"...\"}` without node identifier property.\n            It is an error if the entity exists already with a node identifier for the recursive hierarchy."
+          },
+          {
+            "$Name": "Parent",
+            "$Type": "Edm.PrimitiveType",
+            "$Nullable": true,
+            "@Core.Description": "Node identifier of the new node's parent (null if the new node shall be a root)"
+          },
+          {
+            "$Name": "RightSibling",
+            "$Type": "Edm.PrimitiveType",
+            "$Nullable": true,
+            "@Core.Description": "Node identifier of the new node's right sibling (null if the new node shall be the rightmost sibling)",
+            "@Core.LongDescription": "In the absence of this parameter, the sibling position is determined by the service.",
+            "@Core.OptionalParameter": {}
+          }
+        ],
+        "$ReturnType": {
+          "$Type": "Edm.EntityType",
+          "@Core.Description": "The new node",
+          "@Core.LongDescription": "The values of the properties declared in the [`RecursiveHierarchy`](#RecursiveHierarchyType) annotation\n            can only be obtained with a subsequent GET request that contains a [`Hierarchy.TopLevels`](#TopLevels) transformation\n            in an `$apply` query option."
+        }
+      }
+    ],
+    "Template_MoveAction": [
+      {
+        "$Kind": "Action",
+        "$IsBound": true,
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for actions that move a node to a new parent and are named in [`RecursiveHierarchy/MoveAction`](#RecursiveHierarchyType)",
+        "$Parameter": [
+          { "$Name": "HierarchyNodes", "$Collection": true, "$Type": "Edm.EntityType" },
+          { "$Name": "HierarchyQualifier", "$Type": "Aggregation.HierarchyQualifier" },
+          {
+            "$Name": "Node",
+            "$Type": "Edm.PrimitiveType",
+            "@Core.Description": "Node identifier of the node to be moved"
+          },
+          {
+            "$Name": "Parent",
+            "$Type": "Edm.PrimitiveType",
+            "$Nullable": true,
+            "@Core.Description": "Node identifier of the node's new parent (null if the node shall become a root)"
+          },
+          {
+            "$Name": "KeepParents",
```

Comment:
We could avoid both KeepParents and RemoveParent if we would accept a 2nd action call to a RemoveAction. Keeping would be the default and removing done via that 2nd call. What do you think?

Reply:
In a hierarchy where only one parent is allowed, keeping as default would not work. Moreover, I don't want two requests for the common case where a node is moved from one parent to another.